src/components/pagination/_pagination.scss

```diff
@@ -237,12 +245,30 @@ $css--helpers: true;
 
     .#{$prefix}--select__item-count {
       border-right: $spacing-4xs solid $ui-03;
-      padding-right: 0.5rem;
     }
 
     .#{$prefix}--select__page-number {
       border-left: $spacing-4xs solid $ui-03;
-      padding-left: 0.5rem;
+    }
+
+    .#{$prefix}--select__item-count .#{$prefix}--select-input {
+      @include breakpoint($pagination-breakpoint) {
+        padding-right: 2rem;
+        margin-right: -0.65rem;
+      }
+    }
+
+    .#{$prefix}--select__page-number .#{$prefix}--select-input {
+      @include breakpoint($pagination-breakpoint) {
+        padding-left: 1rem;
+        padding-right: 2rem;
+      }
+    }
+
+    .#{$prefix}--select__page-number .#{$prefix}--select__arrow {
+      @include breakpoint($pagination-breakpoint) {
+        right: 1.6rem;
```

Comment:
Is there any way we could get around magic numbers for positioning? Just curious where things need to have decimal precision in the pixel value 🤔

Reply:
i haven't removed the magic numbers within the current time constraint, but we can open an issue to investigate/refactor this in a future PR